packages/core/src/GondelComponentRegistry.ts

```diff
@@ -38,28 +45,29 @@ export const componentRegistries: {
   [key: string]: GondelComponentRegistry;
 } = ((window as any).__gondelRegistries = (window as any).__gondelRegistries || {});
 
-export function registerComponent(componentName: string, component: IGondelComponent): void;
 export function registerComponent(
   componentName: string,
-  namespace: string | undefined,
-  component: IGondelComponent
-): void;
-export function registerComponent() {
-  const args = arguments;
-  // The componentName is always the first argument
-  const componentName = args[0] as string;
-  // Use namespace from the second argument or fallback to the default "g" if it is missing
-  const namespace = typeof args[1] === "string" ? args[1] : "g";
-  // The last argument is always the component class
-  let component = args[args.length - 1] as IGondelComponent;
+  component: IGondelComponent,
```

Comment:
switching the order would be a breaking change from:

```
registerCompontent('Button', 'g', class Button { ... });
// to
registerCompontent('Button', class Button { ... }, 'g');
```

I believe `Button` and `g` should be close together.  
This change could separate them by hundreds lines of code.